components/suggest/src/db.rs

```diff
@@ -123,14 +124,29 @@ impl<'a> SuggestDao<'a> {
 
     /// Fetches suggestions that match the given keyword from the database.
     pub fn fetch_by_keyword(&self, keyword: &str) -> Result<Vec<Suggestion>> {
+        let pocket_keyword = match keyword.contains(' ') {
+            true => keyword.to_string(),
+            false => keyword.to_owned() + " ",
+        };
         self.conn.query_rows_and_then_cached(
-            "SELECT s.id, k.rank, s.title, s.url, s.provider
+            "SELECT s.id, k.rank, s.title, s.url, s.provider, false as is_top_pick
              FROM suggestions s
              JOIN keywords k ON k.suggestion_id = s.id
              WHERE k.keyword = :keyword
+             UNION
+             SELECT s.id, k.rank, s.title, s.url, s.provider, true as is_top_pick
+             FROM suggestions s
+             JOIN pocket_high_confidence_keywords k ON k.suggestion_id = s.id
+             WHERE k.keyword = :keyword
+             UNION
```

Comment:
If it already finds a record from the high confidence keywords table, doing another lookup against the low-confidence keywords table is unnecessary. Looks like it's not easy to inject that conditional query logic inside this query, guess we will live with that.

Reply:
I bet there's a way to do it, maybe:

```
UNION
SELECT IFNULL(
  (
    SELECT s.id, k.rank, s.title, s.url, s.provider, true as is_top_pick
    FROM suggestions s
    JOIN pocket_high_confidence_keywords k ON k.suggestion_id = s.id
    WHERE k.keyword = :keyword
  ),
  (
    SELECT s.id, k.rank, s.title, s.url, s.provider, false as is_top_pick
    FROM suggestions s
    JOIN pocket_low_confidence_keywords k ON k.suggestion_id = s.id
    WHERE k.keyword BETWEEN :pocket_keyword AND :pocket_keyword || x'ffff'
  )
)
```

The high and low keywords could also be stored in the same table, so you do one lookup to get the matching keyword and its type, whether it's high or low, and then a join to get the suggestion.

Reply:
Good point, but unfortunately, both `IFNULL` and `COALESCE` in SQLite only take scalars as the arguments, selecting multiple columns will lead to a syntax error.